cyberdrop_dl/crawlers/mydesi.py

```diff
@@ -0,0 +1,85 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, ClassVar
+
+from cyberdrop_dl.crawlers.crawler import Crawler, SupportedPaths
+from cyberdrop_dl.data_structures import Resolution
+from cyberdrop_dl.data_structures.url_objects import AbsoluteHttpURL
+from cyberdrop_dl.utils import css
+from cyberdrop_dl.utils.utilities import error_handling_wrapper
+
+if TYPE_CHECKING:
+    from bs4 import BeautifulSoup
+
+    from cyberdrop_dl.data_structures.url_objects import ScrapeItem
+
+
+PRIMARY_URL = AbsoluteHttpURL("https://lolpol.com")
+
+
+class MyDesiCrawler(Crawler):
+    SUPPORTED_DOMAINS = "fry99.com", "lolpol.com", "mydesi.net"
+    SUPPORTED_PATHS: ClassVar[SupportedPaths] = {"Videos": "/title", "Search": "/search/"}
+    PRIMARY_URL: ClassVar[AbsoluteHttpURL] = PRIMARY_URL
+    DOMAIN: ClassVar[str] = "mydesi.net"
+    FOLDER_DOMAIN: ClassVar[str] = "MyDesi"
+
+    async def fetch(self, scrape_item: ScrapeItem) -> None:
+        if scrape_item.url.parts[1] == "search":
+            return await self.search(scrape_item)
+        if len(scrape_item.url.parts) == 2:
+            return await self.video(scrape_item)
+        raise ValueError
+
+    @error_handling_wrapper
+    async def video(self, scrape_item: ScrapeItem) -> None:
+        if await self.check_complete_from_referer(scrape_item):
+            return
+        soup = await self.request_soup(scrape_item.url)
+        resolution, download_url = self.select_highest_quality_video(soup)
+        title = soup.select_one("div.col-12.col-md-8.col-left > h1").text.strip()
+        filename, ext = self.get_filename_and_ext(download_url.name)
+
+        metadata_script = css.get_json_ld(soup)
+        upload_date = metadata_script.get("subjectOf", {}).get("uploadDate", "")
+        scrape_item.possible_datetime = self.parse_date(upload_date)
+        custom_filename = self.create_custom_filename(title, ext, resolution=resolution)
+        return await self.handle_file(download_url, scrape_item, filename, ext, custom_filename=custom_filename)
+
+    @error_handling_wrapper
+    async def search(self, scrape_item: ScrapeItem) -> None:
+        album_initialized: bool = False
+        async for soup in self.web_pager(scrape_item.url, next_page_selector=self.paginate):
+            if not album_initialized:
+                title = self.create_title(soup.select_one("title").text.split("“")[1].split("”")[0])
+                scrape_item.setup_as_album(title)
+                album_initialized = True
```

Comment:
Here you can just grab the search query from the URL itself

`init`page` comes from the URL as well

```suggestion
    async def search(self, scrape_item: ScrapeItem, query: str, init_page: int = 1) -> None:
        title = self.create_title(f"{query} [search]")
        scrape_item.setup_as_album(title)
        
        for page in itertools.count(init_page):
            soup = await self.request_soup(scrape_item.url.with_name(str(page))):
                
```